src/models/searchservice/common/NumberRangeValue.ts

```diff
@@ -0,0 +1,56 @@
+import { Matcher } from './Matcher';
+
+/**
+ * Matcher values of a lower limit for a filter with a $between {@link Matcher}.
+ *
+ * @public
+ */
+export type LowerNumberRangeMatcher = Matcher.GreaterThan | Matcher.GreaterThanOrEqualTo;
+
+/**
+ * Matcher values of an upper limit for a filter with a $between {@link Matcher}.
+ *
+ * @public
+ */
+export type UpperNumberRangeMatcher = Matcher.LessThan | Matcher.LessThanOrEqualTo;
+
+
+/**
+ * The start limit of {@link NumberRangeValue}.
+ *
+ * @public
+ */
+export interface LowerNumberRangeLimit {
+  /** {@link Matcher} for the start limit */
+  matcher: LowerNumberRangeMatcher,
+  /** Value of the limit. */
+  value: number
+}
+
+/**
+ * The end limit of {@link NumberRangeValue}.
+ *
+ * @public
+ */
+export interface UpperNumberRangeLimit {
+  /** {@link Matcher} for the end limit */
+  matcher: UpperNumberRangeMatcher,
+  /** Value of the limit. */
+  value: number
+}
+
+/**
+ * A filter value for a filter with a $between {@link Matcher}.
+ *
+ * @public
+ */
+export interface NumberRangeValue {
+  /** Start limit of the number range value. */
+  start?: LowerNumberRangeLimit
+  /** End limit of the number range value. */
+  end?: UpperNumberRangeLimit
+}
+
+export function isNumberRangeValue(data: unknown): data is NumberRangeValue {
```

Comment:
I think we can put this type guard inside serializeStaticFilters? I kind of like putting it here centralizes it, but we also normally don't put non type/interface code inside the models folder. We're only using it in a single place right now, and maybe we can centralize if we need it later?

Reply:
It was originally in `serializeStaticFilters` but connor suggested it to move to here so it's easier to be consistent with NumberRangeValue if there were any changes to it (refer to the comment made above). I don't have a strong preference here

Reply:
Oh omg I missed the comment. We can keep it! The placement isn't crucial